math/complex_numbers.cpp

```diff
@@ -0,0 +1,193 @@
+     * @param y The imaginary value of the complex number.
+     */
+    Complex(double x, double y) {
+        this->re = x;
+        this->im = y;
+    }
+
+    /**
+     * Complex Constructor which initialises the complex number with no
+     * arguments.
+     */
+    Complex() {
+        this->re = 0.0;
+        this->im = 0.0;
+    }
+
+    /**
+     * Member function (getter) to access the class' re value.
+     */
+    double real() const { return this->re; }
+
+    /**
+     * Member function (getter) to access the class' im value.
+     */
+    double imag() const { return this->im; }
+
+    /**
+     * Member function to which gives the absolute value (modulus) of our
+     * complex number
+     * @return \f$ \sqrt{z \dot \bar{z}} \f$ where \f$ z \f$ is our complex
+     * number.
+     */
+    double abs() const {
+        return std::sqrt(this->re * this->re + this->im * this->im);
+    }
+
+    /**
+     * Operator overload to be able to add two complex numbers.
+     * @param other The other number that is added to the current number.
+     * @return result current number plus other number
+     */
+    Complex operator+(const Complex &other) {
+        Complex result(this->re + other.re, this->im + other.im);
+        return result;
+    }
+
+    /**
+     * Operator overload to be able to subtract two complex numbers.
+     * @param other The other number being subtracted from the current number.
+     * @return result current number subtract other number
+     */
+    Complex operator-(const Complex &other) {
+        Complex result(this->re - other.re, this->im - other.im);
+        return result;
+    }
+
+    /**
+     * Operator overload to be able to multiple two complex numbers.
+     * @param other The other number to multiply the current number to.
+     * @return result current number times other number.
+     */
+    Complex operator*(const Complex &other) {
+        Complex result(this->re * other.re - this->im * other.im,
+                       this->re * other.im + this->im * other.re);
+        return result;
+    }
+
+    /**
+     * Operator overload of the BITWISE NOT which gives us the conjugate of our
+     * complex number. NOTE: This is overloading the BITWISE operator but its
+     * not a BITWISE operation in this definition.
+     * @return result The conjugate of our complex number.
+     */
+    Complex operator~() const {
+        Complex result(this->re, -(this->im));
+        return result;
+    }
+
+    /**
+     * Operator overload to be able to divide two complex numbers. This function
+     * would throw an exception if the other number is zero.
+     * @param other The other number we divide our number by.
+     * @return result Current number divided by other number.
+     */
+    Complex operator/(const Complex &other) {
+        Complex result = *this * ~other;
+        double denominator = other.abs() * other.abs();
+        if (denominator != 0) {
```

Comment:
1. when using doubles, check for `std::numeric_limits<double>::epsilon()` instead of '0'. This is defined in `<limits>` header.
2. when checking the result with output from `std::complex` class, do not check for equality but check for approximately equal to. refer [this example code](https://github.com/TheAlgorithms/C-Plus-Plus/blob/27a63f54275f14b68e96b1ddea075fd4c29e4f51/numerical_methods/brent_method_extrema.cpp#L178)

Reply:
I tweaked my implementations to coincide with the standard libraries to make sure they are equal but I can change it to work with epsilon as well if that is what's preferred😄. Let me know

Reply:
Also which c++ standard is being used in CI workflow? There are some very cool features in C++20 like [this](https://stackoverflow.com/a/57285400)

Reply:
We are currently using C++11 only because if people want to use on older compilers or on embedded platforms like a microcontroller, they still can. Most high-level additions in post C++11 are not supported by the ARM cortex or Tensilica, etc processors.

Also, this is something I suggest but feel free to bring it up in the feature request/issues discussions.